pkg/nmi/server/endpointpolicy.go

```diff
@@ -38,9 +38,13 @@ func ApplyEndpointRoutePolicy(podIP string, metadataIP string, metadataPort stri
 
 	if err != nil {
 		if endpointPolicyError, ok := err.(*endpointPolicyError); ok {
-			return fmt.Errorf("Get endpoint for Pod IP - %s. Error: %w", podIP, endpointPolicyError.err)
+			if endpointPolicyError.errType == InvalidOperation {
+				return fmt.Errorf("Get endpoint for Pod IP - %s. Error: %w", podIP, endpointPolicyError.err)
+			} else if endpointPolicyError.errType == NotFound {
```

Comment:
When getEndpointByIp() return error because some endpoint can't be found, we will return no error here, just point out no endpoint found for that pod ip.

Reply:
Good fix. In this case, I think we can refactor the code a little by reducing the duplicate code

Reply:
Thanks for suggestion, but I can't find an elegant way to do so. I tried to put line 37-49 (line 65-77) to a single function. It should look like:
```
func newFunc(...) (endpoint, error)
```
The problem is I don't know if I should return or continue if the error is nil. Because in line 45, it return nil, so the newFunc() should return nil as error. In such case, the code shouldn't continue executing, but I can't find a way to do so.